index.js

```diff
@@ -15,96 +15,131 @@ function create(win, opts) {
+						electron.clipboard.writeBookmark(props.selectionText);
+					} else {
+						electron.clipboard.writeText(props.selectionText);
+					}
+
+					win.webContents.delete();
+				}
+			}),
+			COPY: decorateMenuItem({
+				id: 'copy',
+				label: 'Copy',
+				enabled: can('Copy'),
+				visible: props.isEditable || hasText,
+				click(menuItem) {
+					props.selectionText = menuItem.transform ? menuItem.transform(props.selectionText) : props.selectionText;
+
+					if (process.platform === 'darwin') {
+						electron.clipboard.writeBookmark(props.selectionText);
+					} else {
+						electron.clipboard.writeText(props.selectionText);
+					}
+				}
+			}),
+			PASTE: decorateMenuItem({
+				id: 'paste',
+				label: 'Paste',
+				enabled: editFlags.canPaste,
+				visible: props.isEditable,
+				click(menuItem) {
+					let clipboardContent;
+					if (process.platform === 'darwin') {
+						clipboardContent = electron.clipboard.readBookmark();
+					} else {
+						clipboardContent = electron.clipboard.readText(props.selectionText);
+					}
+
+					clipboardContent = menuItem.transform ? menuItem.transform(clipboardContent) : clipboardContent;
+
+					win.webContents.insertText(clipboardContent);
+				}
+
+			}),
+			INSPECT: decorateMenuItem({
+				id: 'inspect',
+				label: 'Inspect Element',
+				click() {
+					win.inspectElement(props.x, props.y);
+
+					if (webContents(win).isDevToolsOpened()) {
+						webContents(win).devToolsWebContents.focus();
+					}
+				}
+			}),
+			SEPARATOR: decorateMenuItem({
 				type: 'separator'
-			}, {
+			}),
+			SAVE_IMAGE: decorateMenuItem({
 				id: 'save',
 				label: 'Save Image',
-				click() {
+				visible: props.mediaType === 'image',
+				click(menuItem) {
+					props.srcURL = menuItem.transform ? menuItem.transform(props.srcURL) : props.srcURL;
 					download(win, props.srcURL);
 				}
-			}, {
-				type: 'separator'
-			}];
-		}
-
-		if (props.linkURL && props.mediaType === 'none') {
-			menuTpl = [{
-				type: 'separator'
-			}, {
+			}),
+			COPY_LINK: decorateMenuItem({
 				id: 'copyLink',
 				label: 'Copy Link',
-				click() {
+				visible: props.linkURL.length !== 0 && props.mediaType === 'none',
+				click(menuItem) {
+					props.linkURL = menuItem.transform ? menuItem.transform(props.linkURL) : props.linkURL;
+
 					if (process.platform === 'darwin') {
 						electron.clipboard.writeBookmark(props.linkText, props.linkURL);
 					} else {
 						electron.clipboard.writeText(props.linkURL);
 					}
 				}
-			}, {
-				type: 'separator'
-			}];
+			})
+		};
+
+		const defaultMenu = [defaultActions.SEPARATOR(), defaultActions.CUT(), defaultActions.COPY(), defaultActions.PASTE(), defaultActions.SEPARATOR()];
+
+		let menuTpl = defaultMenu;
+
+		if (opts.menu) {
+			menuTpl = opts.menu(defaultActions, props, win);
+		}
+
+		if (!opts.menu && props.mediaType === 'image') {
+			menuTpl = [defaultActions.SEPARATOR(), defaultActions.SAVE_IMAGE(), defaultActions.SEPARATOR()];
```

Comment:
This logic should be in the save image action, not here. Same with the below for "copy link".